internal/net/grpc/interceptor/server/logging/accesslog.go

```diff
@@ -0,0 +1,145 @@
+//
+// Copyright (C) 2019-2021 vdaas.org vald team <vald@vdaas.org>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package logging provides gRPC interceptors for access logging
+package logging
+
+import (
+	"context"
+	"path"
+	"time"
+
+	"github.com/vdaas/vald/internal/log"
+	"github.com/vdaas/vald/internal/observability/trace"
+	"google.golang.org/grpc"
+)
+
+type AccessLogEntity struct {
+	Grpc      *AccessLogGrpcEntity `json:"grpc,omitempty" yaml:"grpc"`
+	StartTime float64              `json:"startTime,omitempty" yaml:"startTime"`
+	Latency   float64              `json:"latency,omitempty" yaml:"latency"`
+	TraceID   string               `json:"traceID,omitempty" yaml:"traceID"`
+	Error     error                `json:"error,omitempty" yaml:"error"`
+}
+
+type AccessLogGrpcEntity struct {
+	Kind    string `json:"kind,omitempty" yaml:"kind"`
+	Service string `json:"service,omitempty" yaml:"service"`
+	Method  string `json:"method,omitempty" yaml:"method"`
+}
+
+func AccessLogInterceptor() grpc.UnaryServerInterceptor {
+	return func(
+		ctx context.Context,
+		req interface{},
+		info *grpc.UnaryServerInfo,
+		handler grpc.UnaryHandler,
+	) (resp interface{}, err error) {
+		var traceID string
+
+		span := trace.FromContext(ctx)
+		if span != nil {
+			traceID = span.SpanContext().TraceID.String()
+		}
+
+		start := time.Now()
+
+		resp, err = handler(ctx, req)
+
+		latency := float64(time.Since(start)) / float64(time.Second)
```

Comment:
I think it would be better to store time as unix nano second scale.
then we can calculate it on the monitoring query

Reply:
okay i'll revise it.

Reply:
thank you!